draft-ietf-quic-tls.md

```diff
@@ -1058,6 +1058,10 @@ validation was not requested originally.  In such cases, the cookie extension
 could either be absent or it could indicate that an address validation token is
 not present.
 
+A server can use the cookie extension to store all state necessary to continue
+the connection.  This allows a server to avoid committing state for clients that
+have unvalidated source addresses.
+
```

Comment:
I think that I'll leave the tricks for another PR.  I'm not convinced that the trick will work.  It will always require at least two packets because NST and CONNECTION_CLOSE require packet protection (I suppose you might argue that is more reason to add a new cleartext+encrypted packet type).  Then there is the size of the certificate.  It's more computation, more amplification factor, and all we get is one round trip less delay.